tensorflow/python/data/experimental/kernel_tests/unique_test.py

```diff
@@ -76,6 +79,36 @@ def testSimpleString(self):
         (["foo", "bar", "baz", "baz", "bar", "foo"], ["foo", "bar", "baz"]),
     ])
 
+  @combinations.generate(test_base.graph_only_combinations())
+  def testTypeMismatch(self):
+
+    # raises InternalError when dtypes don't match.
+    # NOTE: Generating the following expected outputs can be considered/taken up as an
```

Comment:
In TensorFlow, tensors have a specific dtype which is not allowed to change. I think it makes sense to validate the behavior of raising an error when the types don't match. But having the type change mid-dataset isn't something we plan to support, so we shouldn't have a test suggesting such a plan.

Reply:
Will make the necessary changes and commit. Thanks for the info.